### PR TITLE
[5.0] Add symlink option to PublishCommand

### PR DESCRIPTION
**Pull request by barryvdh**
This is an alternative for #7168 This doesn't mean I favor this one, it's just a different approach.

This adds a symlink option to create relative symlinks to folders, instead of copying them.

Advantage is that the assets aren't duplicated. This is especially useful for packages relying on js/css from external packages.

Usage would be just `php artisan vendor:publish --tag=assets --symlink` for example.

If the symlink doesn't work, the files are copied like before.

The difference with the other PR is that that adds a new command, so it depends what becomes the preferred way of publishing assets that aren't meant to be modified by developers.

### CODE REVIEW

File: src/Illuminate/Foundation/Console/VendorPublishCommand.php [hunk 1]
```diff
@@ -116,6 +121,43 @@ protected function publishDirectory($from, $to)
 	}
 
 	/**
+	 * Symlink the directory to the given directory.
+	 *
+	 * @param  string  $from
+	 * @param  string  $to
+	 * @return boolean
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/VendorPublishCommand.php]: bool

Comment (barryvdh) [inline src/Illuminate/Foundation/Console/VendorPublishCommand.php]: Fixed

**Comment by GrahamCampbell** [comment]
Closing as #7168 was rejected for now.

**Comment by barryvdh** [comment]
Why? This is different

**Comment by GrahamCampbell** [comment]
It implements the same idea. It was the idea Taylor was opposed to, not the implementation. I had a short chat with him about it.

**Comment by barryvdh** [comment]
Care to elaborate?
And this is why I created a proposal in the first place :(

**Comment by GrahamCampbell** [comment]
We both don't think it's necessary and would probably be better as a third party package. It's possible that we merge this into 5.1, but I don't see the point in that if we move to puli in 5,1 to deal with this sort of thing.

**Comment by GrahamCampbell** [comment]
I don't think you're time's been wasted, because you can still turn this into a package if it's useful for you.

**Comment by barryvdh** [comment]
Yeah well the idea was to have 1 good way to handle the assets in L5, not create my own package /command that most packages wouldn't use. But I'll see if I can extend the default package publisher or something.

Anyways, I can understand not merging this but I'd just appreciate it if the communication was a little better. Instead of asking me to create a PR and then shooting it down, perhaps you could have chatted about it before that.

**Comment by GrahamCampbell** [comment]
Sorry. I guess I didn't really think through whether we really needed this or not properly at the time.

**Comment by JosephSilber** [comment]
@GrahamCampbell It's not just about this particular case. It's the way issues are handled in general.

Taylor has said that the Github issues are the place to discuss details about feature implementations (in addition to #laravel-dev). Lately a lot of issues have been closed as "Just create a PR for it".

Are issues only meant to discuss mega changes, but smaller ones (that take < 2 hours to implement) are only OK as a PR? I'd say thats a bit harsh.

Can we get clarification on this?

**Comment by barryvdh** [comment]
Yeah and it isn't that writing the takes one or two hours, it's also about testing the code and finding the best way to do it. And if you just write some code, there is a big chance that you, Taylor or someone else wants to do it differently and you can redo a lot. And also in this case, the functionality changed so it was relevant or a different implementation was possible.

It just would make it easier for everyone if we could discuss proposals and how it should be implemented. I thing 2 hours for an initial PR is a lot to do without a proposal..

And I understand that the timeframe of the L5  is pressing, but just closing issues doesnt really help the community..

**Comment by GrahamCampbell** [comment]
@JosephSilber I'm not sure what other option we have. You say they're there for discussion, but the reality is issues will often sit there untouched for months. Starting 6 weeks ago, I've changed my style so that things get dealt with more quickly.

Taylor did disable the issues completely a while back, and even said he'd do it again if there was a way to have the old issues still showing, but just let people create issues.

**Comment by franzliedke** [comment]
They're not being dealt with, though. Just closed.

**Comment by GrahamCampbell** [comment]
> They're not being dealt with, though. Just closed.

@franzliedke It's been rejected by Taylor. We're now done here. People may continue to discuss this on the closed issue if they want to, but closing it means that it's done from the point of me looking through the current issues.

**Pull request closed without merging**